File: leetcode3/최원준/1967. Number of Strings That Appear as Substrings in Word.py
'''
1. 아이디어 :
n, m이 최대 100이기떄문에 n**2으로 쉽게 풀 수 있습니다.

2. 시간복잡도 :
    O(n * n)

3. 자료구조/알고리즘 :
-
'''

class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        ans = 0
        for p in patterns:
            if p in word:
                ans+=1
        return ans

File: leetcode3/이진희/1967_Number of Strings That Appear as Substrings in Word.java
/*

1. 아이디어 : 각 pattern들이 word의 부분문자열인지 파악

2. 시간복잡도 : O(patterns.length*N*M) (각 비교 문자열의 길이 곱)

3. 자료구조/알고리즘 : 문자열

 */

class Solution {
    public int numOfStrings(String[] patterns, String word) {
        int ans = 0;
        for(String s : patterns) {
            if(word.contains(s)) ans++; 
        }

        return ans;
    }
}